src/group_theory/torsion.lean

```diff
@@ -54,35 +60,84 @@ noncomputable def is_torsion.group [monoid G] (tG : is_torsion G) : group G :=
   end,
   ..‹monoid G› }
 
+section group
+
 variables [group G] {N : subgroup G}
 
-/--Subgroups of torsion groups are torsion groups. -/
+/-- Subgroups of torsion groups are torsion groups. -/
 @[to_additive "Subgroups of additive torsion groups are additive torsion groups."]
 lemma is_torsion.subgroup (tG : is_torsion G) (H : subgroup G) : is_torsion H :=
 λ h, (is_of_fin_order_iff_coe _ h).mpr $ tG h
 
-/--Quotient groups of torsion groups are torsion groups. -/
+/-- Quotient groups of torsion groups are torsion groups. -/
 @[to_additive "Quotient groups of additive torsion groups are additive torsion groups."]
 lemma is_torsion.quotient_group [nN : N.normal] (tG : is_torsion G) : is_torsion (G ⧸ N) :=
 λ h, quotient_group.induction_on' h $ λ g, (tG g).quotient N g
 
-/--If a group exponent exists, the group is torsion. -/
-@[to_additive exponent_exists.is_add_torsion]
+/-- If a group exponent exists, the group is torsion. -/
+@[to_additive exponent_exists.is_add_torsion
+  "If a group exponent exists, the group is additively torsion."]
 lemma exponent_exists.is_torsion (h : exponent_exists G) : is_torsion G := begin
   obtain ⟨n, npos, hn⟩ := h,
   intro g,
   exact (is_of_fin_order_iff_pow_eq_one g).mpr ⟨n, npos, hn g⟩,
 end
 
-/--The group exponent exists for any bounded torsion group. -/
-@[to_additive is_add_torsion.exponent_exists]
+/-- The group exponent exists for any bounded torsion group. -/
+@[to_additive is_add_torsion.exponent_exists
+  "The group exponent exists for any bounded additive torsion group."]
 lemma is_torsion.exponent_exists
   (tG : is_torsion G) (bounded : (set.range (λ g : G, order_of g)).finite) :
   exponent_exists G :=
 exponent_exists_iff_ne_zero.mpr $
   (exponent_ne_zero_iff_range_order_of_finite (λ g, order_of_pos' (tG g))).mpr bounded
 
-/--Finite groups are torsion groups.-/
-@[to_additive is_add_torsion_of_fintype]
+/-- Finite groups are torsion groups. -/
+@[to_additive is_add_torsion_of_fintype "Finite additive groups are additive torsion groups."]
 lemma is_torsion_of_fintype [fintype G] : is_torsion G :=
 exponent_exists.is_torsion $ exponent_exists_iff_ne_zero.mpr exponent_ne_zero_of_fintype
+
+end group
+
+section comm_group
+
+variables (G) [comm_group G]
+
+/-- The torsion subgroup of an abelian group. -/
+@[to_additive add_tor "The torsion subgroup of an additive abelian group."]
```

Comment:
Can you add a similar def for monoids and the fact that `torsion (G.to_monoid) = (torsion G).to_monoid`? Thanks!

Reply:
Done I believe, and Eric's suggestion fixed `simps`.